### PR TITLE
BUG: Convert float freqstrs to ints at finer resolution

### LINKED ISSUE #8419: date_range treats '.5s' as '5s'
I'm new to pandas, so maybe I'm missing something, but I'd expected freq='.5s' to give me 500ms intervals, not 5s intervals:

> > > pd.date_range(0, periods=5, freq='.5s')[1]
> > > Timestamp('1970-01-01 00:00:05', offset='5S')
> > > pd.date_range(0, periods=5, freq='500ms')[1]
> > > Timestamp('1970-01-01 00:00:00.500000', offset='500L')
> > > pd.**version**
> > > '0.14.1'
> > > import sys; sys.version
> > > '3.4.1 (default, Jul  6 2014, 20:01:46) \n[GCC 4.9.0]'

Behaviour is the same under Python 2 and 3.

### PR DESCRIPTION
**Pull request by jisantuc**
- [x] closes #8419 
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

Passing `'0.5min'` as a frequency string should generate 30 second
intervals, rather than five minute intervals. By recursively increasing
resolution until one is found for which the frequency is an integer,
this commit ensures that that's the case for resolutions from days to
microsecond

### CODE REVIEW

File: pandas/tseries/tests/test_frequencies.py [hunk 1]
```diff
@@ -379,6 +380,17 @@ def test_freq_to_reso(self):
             result = Reso.get_freq(Reso.get_str(Reso.get_reso_from_freq(freq)))
             self.assertEqual(freq, result)
 
+    def test_resolution_bumping(self):
+        Reso = frequencies.Resolution
+
+        self.assertEqual(Reso.bump_to_int(1.5, 'T'), (90, 'S'))
+        self.assertEqual(Reso.bump_to_int(62.4, 'T'), (3744, 'S'))
```
Comment (jreback) [inline pandas/tseries/tests/test_frequencies.py]: can u exercise some more fail cases
and systematically cover cases that should succeed

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tseries/frequencies.py [hunk 2]
```diff
@@ -160,6 +181,34 @@ def get_reso_from_freq(cls, freq):
...
+        >>> Resolution.bump_to_int(1, 'D')
+        (1, 'D')
+        """
+
+        if np.isclose(value % 1, 0):
+            return int(value), freq
+        else:
+            start_reso = cls.get_reso_from_freq(freq)
+            if start_reso == 0:
+                raise ValueError(_CANNOT_CONVERT_TO_INT)
```
Comment (sinhrks) [inline pandas/tseries/frequencies.py]: pls move error message here, because the message is used only once.

File: pandas/tseries/frequencies.py [hunk 3]
```diff
@@ -63,7 +63,28 @@ class Resolution(object):
...
+        RESO_DAY: 24
+    }
+
+    _reso_str_bump_map = {
+        'D': 'H',
+        'H': 'T',
+        'T': 'S',
+        'S': 'L',
+        'L': 'U',
+        'U': None
```
Comment (sinhrks) [inline pandas/tseries/frequencies.py]: Pls add Nanosecond.

File: pandas/tseries/tests/test_frequencies.py (hunk 1 above)
Comment (sinhrks) [inline pandas/tseries/tests/test_frequencies.py]: Pls also add some failure cases with invalid str, such as ``1.5.0S``, ``1.S``.

File: pandas/tseries/tests/test_frequencies.py [hunk 4]
```diff
@@ -108,10 +113,6 @@ def test_to_offset_invalid(self):
         with tm.assertRaisesRegexp(ValueError, 'Invalid frequency: -2D:3H'):
             frequencies.to_offset('-2D:3H')
 
-        # ToDo: Must be fixed in #8419
-        with tm.assertRaisesRegexp(ValueError, 'Invalid frequency: .5S'):
-            frequencies.to_offset('.5S')
```
Comment (sinhrks) [inline pandas/tseries/tests/test_frequencies.py]: Can you change it to compare with 500 milisec.

File: doc/source/whatsnew/v0.19.1.txt [hunk 5]
```diff
@@ -30,3 +30,5 @@ Performance Improvements
 
 Bug Fixes
 ~~~~~~~~~
+
+- Bug in date range treating e.g. 0.5s as 5s (:issue:`8419`)
```
Comment (sinhrks) [inline doc/source/whatsnew/v0.19.1.txt]: can you add occurrence condition, like

- Bug in multiple offset aliases with decimal points is regarded as int (e.g. 0.5s as 5s).

File: pandas/tseries/frequencies.py [hunk 6]
```diff
@@ -160,6 +181,34 @@ def get_reso_from_freq(cls, freq):
         """
         return cls.get_reso(cls.get_str_from_freq(freq))
 
+    @classmethod
+    def bump_to_int(cls, value, freq):
```
Comment (sinhrks) [inline pandas/tseries/frequencies.py]: can the method name be clearler, like``get_stride_from_decimal``

File: pandas/tseries/frequencies.py [hunk 7]
```diff
@@ -160,6 +181,34 @@ def get_reso_from_freq(cls, freq):
...
+        """
+        Increase resolution an integer value is available to create an
```
Comment (sinhrks) [inline pandas/tseries/frequencies.py]: Convert freq with decimal stride to higher freq with integer stride.

File: pandas/tseries/tests/test_frequencies.py (hunk 4 above)
Comment (sinhrks) [inline pandas/tseries/tests/test_frequencies.py]: Also pls add some tests whether this works with combined offset, like ``.5H.5S``.

Comment (jisantuc) [inline pandas/tseries/tests/test_frequencies.py]: `1.S` is valid now, parsed as `1 second`. Should it not be?

File: pandas/tseries/frequencies.py [hunk 8]
```diff
@@ -38,18 +38,20 @@ class FreqGroup(object):
     FR_NS = 12000
 
 
-US_RESO = 0
-MS_RESO = 1
-S_RESO = 2
-T_RESO = 3
-H_RESO = 4
-D_RESO = 5
+NS_RESO = 0
```
Comment (jreback) [inline pandas/tseries/frequencies.py]: so would be nice to clean this up. This is essentially an Enum, but that only exists in 3.5 (maybe we could add the backport of https://pypi.python.org/pypi/enum34 as an in-inline package), not sure this is worth it though.

In any event just put these in a list and ``enumerate`` if needed.

File: pandas/tseries/tests/test_frequencies.py (hunk 4 above)
Comment (sinhrks) [inline pandas/tseries/tests/test_frequencies.py]: You're right. should be 1 sec like ``float("1.")`` is valid.

File: pandas/tseries/frequencies.py (hunk 8 above)
Comment (jisantuc) [inline pandas/tseries/frequencies.py]: Just learned they need to be available at the module level for some of the c imports, but inside the class they can live in a list, so I did that.

File: doc/source/whatsnew/v0.19.1.txt [hunk 9]
```diff
@@ -30,3 +30,5 @@ Performance Improvements
 
 Bug Fixes
 ~~~~~~~~~
+
+- Bug in multiple offset aliases with decimal points regarded as ints (e.g. 0.5s as 5s) (:issue:`8419`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.19.1.txt]: Can you move this to v0.20.0 file and add it to the enhancements section? It was a bug that is was parsed incorrectly and did not error, but supporting fractional resolutions is an enhancement that can be listed in the enhancement section

File: pandas/tseries/frequencies.py [hunk 10]
```diff
@@ -38,32 +38,52 @@ class FreqGroup(object):
...
 
     # defined in period.pyx
     # note that these are different from freq codes
-    RESO_US = US_RESO
-    RESO_MS = MS_RESO
-    RESO_SEC = S_RESO
-    RESO_MIN = T_RESO
-    RESO_HR = H_RESO
-    RESO_DAY = D_RESO
+    RESOS = [RESO_NS, RESO_US, RESO_MS, RESO_SEC, RESO_MIN, RESO_HR, RESO_DAY]
```
Comment (jorisvandenbossche) [inline pandas/tseries/frequencies.py]: is this used somewhere?

**Review by jorisvandenbossche — COMMENTED** [review]
Can you rebase and update?

File: pandas/tseries/frequencies.py (hunk 10 above)
Comment (jisantuc) [inline pandas/tseries/frequencies.py]: Nope, will destruct.

File: pandas/tseries/frequencies.py (hunk 8 above)
Comment (jisantuc) [inline pandas/tseries/frequencies.py]: They're used in some of the `@classmethod`s it turns out so need to stay for now

**Comment by jreback** [comment]
can you rebase

File: pandas/tseries/tests/test_frequencies.py [hunk 11]
```diff
@@ -379,6 +402,25 @@ def test_freq_to_reso(self):
             result = Reso.get_freq(Reso.get_str(Reso.get_reso_from_freq(freq)))
             self.assertEqual(freq, result)
 
+    def test_resolution_bumping(self):
+        Reso = frequencies.Resolution
```
Comment (jreback) [inline pandas/tseries/tests/test_frequencies.py]: can you add the issue number here as as comment

File: pandas/tseries/frequencies.py [hunk 12]
```diff
@@ -160,6 +183,36 @@ def get_reso_from_freq(cls, freq):
         """
         return cls.get_reso(cls.get_str_from_freq(freq))
 
+    @classmethod
+    def get_stride_from_decimal(cls, value, freq):
+        """
+        Convert freq with decimal stride into a higher freq with integer stride
```
Comment (jreback) [inline pandas/tseries/frequencies.py]: can you add Parameters/Returns section to detail the args

**Comment by jreback** [comment]
just a little docs, otherwise lgtm.

**Comment by jorisvandenbossche** [comment]
Made the latest changes quickly myself. @jisantuc Thanks a lot for this nice contribution!

**Pull request merged**